### PR TITLE
Add functionality and tests for delete c command

### LINKED ISSUE #40: As a CS student, I can delete company applications that I am no longer interested in
... so that my list of companies becomes less cluttered.

### PR DESCRIPTION
**Pull request by oeggy03**
Update DeleteCommandParser to parse the the delete commands (delete p and delete c)
Add the delete c command.
Add error handling and tests for delete c command.
Update existing tests and functions to accomodate change in delete command format (Previously delete INDEX, now delete c INDEX or delete p INDEX)
Update MESSAGE_USAGE and INVALID COMMAND texts for the delete command.

### CODE REVIEW

**Comment by oeggy03** [comment]
For issue #40

File: src/main/java/seedu/address/logic/commands/deletecommands/DeleteCommand.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+import seedu.address.logic.commands.Command;
+import seedu.address.logic.commands.CommandResult;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+
+/**
+ * Lists all persons or companies in the address book to the user.
+ */
+public abstract class DeleteCommand extends Command {
```
Comment (lunaroddity) [inline src/main/java/seedu/address/logic/commands/deletecommands/DeleteCommand.java:11]: Nice abstraction!

File: src/test/java/seedu/address/logic/commands/DeleteCompanyCommandTest.java [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.Messages;
+import seedu.address.logic.commands.deletecommands.DeleteCompanyCommand;
+import seedu.address.model.Model;
+import seedu.address.model.ModelManager;
+import seedu.address.model.UserPrefs;
+import seedu.address.model.company.Company;
+
+public class DeleteCompanyCommandTest {
```
Comment (lunaroddity) [inline src/test/java/seedu/address/logic/commands/DeleteCompanyCommandTest.java:23]: The tests are extensive, very nicely done :D

**Review by lunaroddity — APPROVED** [review]
LGTM!

**Pull request merged**